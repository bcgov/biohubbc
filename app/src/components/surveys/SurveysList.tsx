import Box from '@material-ui/core/Box';
import Chip from '@material-ui/core/Chip';
import Link from '@material-ui/core/Link';
import { Theme } from '@material-ui/core/styles/createMuiTheme';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
// import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';
// import { handleChangePage, handleChangeRowsPerPage } from 'utils/tablePaginationUtils';
// import { getFormattedDateRangeString } from 'utils/Utils';
import { mdiAlertCircle } from '@mdi/js';
import Icon from '@mdi/react';
import clsx from 'clsx';
// import { DATE_FORMAT } from 'constants/dateTimeFormats';
import { SurveyStatusType } from 'constants/misc';
import { SurveyViewObject } from 'interfaces/useSurveyApi.interface';
import moment from 'moment';
import React, { useState } from 'react';

const useStyles = makeStyles((theme: Theme) => ({
  surveyTable: {
    tableLayout: 'fixed'
  }
}));

export interface ISurveysListProps {
  surveysList: SurveyViewObject[];
  projectId: number;
}

const SurveysList: React.FC<ISurveysListProps> = (props) => {
  const classes = useStyles();

  const [rowsPerPage] = useState(5);
  const [page] = useState(0);

  const getSurveyCompletionStatusType = (surveyObject: SurveyViewObject): SurveyStatusType => {
    if (
      surveyObject.survey_details.end_date &&
      moment(surveyObject.survey_details.end_date).endOf('day').isBefore(moment())
    ) {
      return SurveyStatusType.COMPLETED;
    }

    return SurveyStatusType.ACTIVE;
  };

  const getChipIcon = (status_name: string) => {
    let chipLabel;
    let chipStatusClass;

    if (SurveyStatusType.ACTIVE === status_name) {
      chipLabel = 'Active';
    } else if (SurveyStatusType.COMPLETED === status_name) {
      chipLabel = 'Completed';
    }

    return <Chip color="secondary" style={{ minWidth: '100px' }} className={clsx(chipStatusClass)} label={chipLabel} />;
  };

  return (
    <>
      <TableContainer>
        <Table aria-label="surveys-list-table" className={classes.surveyTable}>
          <TableHead>
            <TableRow>
              <TableCell>Name</TableCell>
              <TableCell>Species</TableCell>
              <TableCell>Purpose</TableCell>
              <TableCell width="220px">Status</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {props.surveysList.length > 0 &&
              props.surveysList.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((row, index) => (
                <TableRow key={index}>
                  <TableCell scope="row">
                    <Link
                      style={{ fontWeight: 'bold' }}
                      underline="always"
                      href={`/admin/projects/${props.projectId}/surveys/${row.survey_details.id}/details`}>
                      {row.survey_details.survey_name}
                    </Link>
                  </TableCell>
                  <TableCell>
                    {[...row.species?.focal_species_names, ...row.species?.ancillary_species_names].join(', ')}
                  </TableCell>
                  <TableCell>Community Composition</TableCell>
                  {/* <TableCell>
                    Call Playback
                  </TableCell> */}
                  {/* <TableCell>
                    {getFormattedDateRangeString(
                      DATE_FORMAT.ShortMediumDateFormat,
                      row.survey_details.start_date,
                      row.survey_details.end_date
                    )}
                  </TableCell> */}
                  <TableCell>
                    <Chip
                      size="small"
                      color="secondary"
                      label="Pending Review"
                      icon={<Icon path={mdiAlertCircle} size={0.8} />}
                    />

                    <Box hidden>{getChipIcon(getSurveyCompletionStatusType(row))}</Box>
                  </TableCell>
                </TableRow>
              ))}
            {!props.surveysList.length && (
              <TableRow>
                <TableCell colSpan={4} align="center">
                  <strong>No Surveys</strong>
                </TableCell>
              </TableRow>
            )}
          </TableBody>
        </Table>
      </TableContainer>
      {/* {props.surveysList.length > 0 && (
        <TablePagination
          rowsPerPageOptions={[5, 10, 15, 20]}
          component="div"
          count={props.surveysList.length}
          rowsPerPage={rowsPerPage}
          page={page}
          onChangePage={(event: unknown, newPage: number) => handleChangePage(event, newPage, setPage)}
          onChangeRowsPerPage={(event: React.ChangeEvent<HTMLInputElement>) =>
            handleChangeRowsPerPage(event, setPage, setRowsPerPage)
          }
        />
      )} */}
    </>
  );
};

export default SurveysList;
