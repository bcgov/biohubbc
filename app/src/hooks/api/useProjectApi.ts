import { AxiosInstance, CancelTokenSource } from 'axios';
import { IReportMetaForm } from 'components/attachments/ReportMetaForm';
import { IEditReportMetaForm } from 'components/attachments/EditReportMetaForm';
import {
  IAddProjectParticipant,
  ICreateProjectRequest,
  ICreateProjectResponse,
  IGetUserProjectsListResponse,
  IGetProjectAttachmentsResponse,
  IGetProjectForUpdateResponse,
  IGetProjectForViewResponse,
  IGetProjectParticipantsResponse,
  IGetProjectsListResponse,
  IGetReportMetaData,
  IProjectAdvancedFilterRequest,
  IUpdateProjectRequest,
  IUploadAttachmentResponse,
  UPDATE_GET_ENTITIES
} from 'interfaces/useProjectApi.interface';
import qs from 'qs';

/**
 * Returns a set of supported api methods for working with projects.
 *
 * @param {AxiosInstance} axios
 * @return {*} object whose properties are supported api methods.
 */
const useProjectApi = (axios: AxiosInstance) => {
  /**
   * Get projects from userId
   *
   * @param {number} userId
   * @return {*} {Promise<IGetProjectsListResponse[]>}
   */
  const getAllUserProjectsForView = async (userId: number): Promise<IGetUserProjectsListResponse[]> => {
    const { data } = await axios.get(`/api/user/${userId}/projects/get`);
    return data;
  };

  /**
   * Get project attachments based on project ID
   *
   * @param {AxiosInstance} axios
   * @returns {*} {Promise<IGetProjectAttachmentsResponse>}
   */
  const getProjectAttachments = async (projectId: number): Promise<IGetProjectAttachmentsResponse> => {
    const { data } = await axios.get(`/api/project/${projectId}/attachments/list`);

    return data;
  };

  /**
   * Delete project based on project ID
   *
   * @param {number} projectId
   * @returns {*} {Promise<boolean>}
   */
  const deleteProject = async (projectId: number): Promise<boolean> => {
    const { data } = await axios.delete(`/api/project/${projectId}/delete`);

    return data;
  };

  /**
   * Delete project attachment based on project and attachment ID
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {string} attachmentType
   * @param {any} securityToken
   * @returns {*} {Promise<number>}
   */
  const deleteProjectAttachment = async (
    projectId: number,
    attachmentId: number,
    attachmentType: string,
    securityToken: any
  ): Promise<number> => {
    const { data } = await axios.post(`/api/project/${projectId}/attachments/${attachmentId}/delete`, {
      attachmentType,
      securityToken
    });

    return data;
  };

  /**
   * Get project attachment S3 url based on project and attachment ID
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {string} attachmentType
   * @return {*}  {Promise<string>}
   */
  const getAttachmentSignedURL = async (
    projectId: number,
    attachmentId: number,
    attachmentType: string
  ): Promise<string> => {
    const { data } = await axios.get(`/api/project/${projectId}/attachments/${attachmentId}/getSignedUrl`, {
      params: { attachmentType: attachmentType },
      paramsSerializer: (params) => {
        return qs.stringify(params);
      }
    });

    return data;
  };

  /**
   * Get projects list (potentially based on filter criteria).
   *
   * @param {IProjectAdvancedFilterRequest} filterFieldData
   * @return {*}  {Promise<IGetProjectsListResponse[]>}
   */
  const getProjectsList = async (
    filterFieldData?: IProjectAdvancedFilterRequest
  ): Promise<IGetProjectsListResponse[]> => {
    const { data } = await axios.post(`/api/projects`, filterFieldData || {});

    return data;
  };

  /**
   * Get project details based on its ID for viewing purposes.
   *
   * @param {number} projectId
   * @return {*} {Promise<IGetProjectForViewResponse>}
   */
  const getProjectForView = async (projectId: number): Promise<IGetProjectForViewResponse> => {
    const { data } = await axios.get(`/api/project/${projectId}/view`);

    return data;
  };

  /**
   * Get project details based on its ID for updating purposes.
   *
   * @param {number} projectId
   * @returns
   */
  const getProjectForUpdate = async (
    projectId: number,
    entities: UPDATE_GET_ENTITIES[]
  ): Promise<IGetProjectForUpdateResponse> => {
    const { data } = await axios.get(`api/project/${projectId}/update`, {
      params: { entity: entities },
      paramsSerializer: (params) => {
        return qs.stringify(params);
      }
    });

    return data;
  };

  /**
   * Update an existing project.
   *
   * @param {number} projectId
   * @param {IUpdateProjectRequest} projectData
   * @return {*}  {Promise<any>}
   */
  const updateProject = async (projectId: number, projectData: IUpdateProjectRequest): Promise<any> => {
    const { data } = await axios.put(`api/project/${projectId}/update`, projectData);

    return data;
  };

  /**
   * Create a new project.
   *
   * @param {ICreateProjectRequest} project
   * @return {*}  {Promise<ICreateProjectResponse>}
   */
  const createProject = async (project: ICreateProjectRequest): Promise<ICreateProjectResponse> => {
    const { data } = await axios.post('/api/project', project);

    return data;
  };

  /**
   * Upload project attachments.
   *
   * @param {number} projectId
   * @param {File} file
   * @param {string} attachmentType
   * @param {CancelTokenSource} [cancelTokenSource]
   * @param {(progressEvent: ProgressEvent) => void} [onProgress]
   * @return {*}  {Promise<string[]>}
   */
  const uploadProjectAttachments = async (
    projectId: number,
    file: File,
    cancelTokenSource?: CancelTokenSource,
    onProgress?: (progressEvent: ProgressEvent) => void
  ): Promise<IUploadAttachmentResponse> => {
    const req_message = new FormData();

    req_message.append('media', file);

    const { data } = await axios.post(`/api/project/${projectId}/attachments/upload`, req_message, {
      cancelToken: cancelTokenSource?.token,
      onUploadProgress: onProgress
    });

    return data;
  };

  /**
   * Upload project reports.
   *
   * @param {number} projectId
   * @param {File} file
   * @param {string} attachmentType
   * @param {CancelTokenSource} [cancelTokenSource]
   * @param {(progressEvent: ProgressEvent) => void} [onProgress]
   * @return {*}  {Promise<string[]>}
   */
  const uploadProjectReports = async (
    projectId: number,
    file: File,
    attachmentMeta: IReportMetaForm,
    cancelTokenSource?: CancelTokenSource,
    onProgress?: (progressEvent: ProgressEvent) => void
  ): Promise<IUploadAttachmentResponse> => {
    const req_message = new FormData();

    req_message.append('media', file);

    if (attachmentMeta) {
      req_message.append('attachmentMeta[title]', attachmentMeta.title);
      req_message.append('attachmentMeta[year_published]', String(attachmentMeta.year_published));
      req_message.append('attachmentMeta[description]', attachmentMeta.description);
      attachmentMeta.authors.forEach((authorObj, index) => {
        req_message.append(`attachmentMeta[authors][${index}][first_name]`, authorObj.first_name);
        req_message.append(`attachmentMeta[authors][${index}][last_name]`, authorObj.last_name);
      });
    }

    const { data } = await axios.post(`/api/project/${projectId}/attachments/report/upload`, req_message, {
      cancelToken: cancelTokenSource?.token,
      onUploadProgress: onProgress
    });

    return data;
  };

  /**
   * Update project attachment metadata.
   *
   * @param {number} projectId
   * @param {string} attachmentType
   * @param {CancelTokenSource} [cancelTokenSource]
   * @param {(progressEvent: ProgressEvent) => void} [onProgress]
   * @return {*}  {Promise<string[]>}
   */
  const updateProjectReportMetadata = async (
    projectId: number,
    attachmentId: number,
    attachmentType: string,
    attachmentMeta: IEditReportMetaForm,
    revisionCount: number
  ): Promise<number> => {
    const obj = {
      attachment_type: attachmentType,
      attachment_meta: {
        title: attachmentMeta.title,
        year_published: attachmentMeta.year_published,
        authors: attachmentMeta.authors,
        description: attachmentMeta.description
      },
      revision_count: revisionCount
    };

    const { data } = await axios.put(`/api/project/${projectId}/attachments/${attachmentId}/metadata/update`, obj);
    return data;
  };

  /**
   * Make security status of project attachment secure.
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {string} attachmentType
   * @return {*}  {Promise<any>}
   */
  const makeAttachmentSecure = async (
    projectId: number,
    attachmentId: number,
    attachmentType: string
  ): Promise<any> => {
    const { data } = await axios.put(`/api/project/${projectId}/attachments/${attachmentId}/makeSecure`, {
      attachmentType
    });

    return data;
  };

  /**
   * Make security status of project attachment unsecure.
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {any} securityToken
   * @param {string} attachmentType
   * @return {*}  {Promise<any>}
   */
  const makeAttachmentUnsecure = async (
    projectId: number,
    attachmentId: number,
    securityToken: any,
    attachmentType: string
  ): Promise<any> => {
    const { data } = await axios.put(`/api/project/${projectId}/attachments/${attachmentId}/makeUnsecure`, {
      securityToken,
      attachmentType
    });

    return data;
  };

  /**
   * Delete funding source based on project and funding source ID
   *
   * @param {number} projectId
   * @param {number} pfsId
   * @returns {*} {Promise<any>}
   */
  const deleteFundingSource = async (projectId: number, pfsId: number): Promise<any> => {
    const { data } = await axios.delete(`/api/project/${projectId}/funding-sources/${pfsId}/delete`);

    return data;
  };

  /**
   * Add new funding source based on projectId
   *
   * @param {number} projectId
   * @returns {*} {Promise<any>}
   */
  const addFundingSource = async (projectId: number, fundingSource: any): Promise<any> => {
    const { data } = await axios.post(`/api/project/${projectId}/funding-sources/add`, fundingSource);

    return data;
  };

  /**
   * Publish/unpublish a project.
   *
   * @param {number} projectId the project id
   * @param {boolean} publish set to `true` to publish the project, `false` to unpublish the project.
   * @return {*}  {Promise<any>}
   */
  const publishProject = async (projectId: number, publish: boolean): Promise<any> => {
    const { data } = await axios.put(`/api/project/${projectId}/publish`, { publish: publish });
    return data;
  };

  /**
   * Get project report metadata based on project ID, attachment ID, and attachmentType
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {string} attachmentType
   * @return {*}  {Promise<IGetReportMetaData>}
   */
  const getProjectReportMetadata = async (projectId: number, attachmentId: number): Promise<IGetReportMetaData> => {
    const { data } = await axios.get(`/api/project/${projectId}/attachments/${attachmentId}/metadata/get`, {
      params: {},
      paramsSerializer: (params) => {
        return qs.stringify(params);
      }
    });

    return data;
  };

  /**
   * Get all project participants.
   *
   * @param {number} projectId
   * @return {*}  {Promise<IGetProjectParticipantsResponse>}
   */
  const getProjectParticipants = async (projectId: number): Promise<IGetProjectParticipantsResponse> => {
    const { data } = await axios.get(`/api/project/${projectId}/participants/get`);

    return data;
  };

  /**
   * Add new project participants.
   *
   * @param {number} projectId
   * @param {IAddProjectParticipant[]} participants
   * @return {*}  {Promise<boolean>} `true` if the request was successful, false otherwise.
   */
  const addProjectParticipants = async (
    projectId: number,
    participants: IAddProjectParticipant[]
  ): Promise<boolean> => {
    const { status } = await axios.post(`/api/project/${projectId}/participants/create`, { participants });

    return status === 200;
  };

  /**
   * Remove existing project participant.
   *
   * @param {number} projectId
   * @param {number} projectParticipationId
   * @return {*}  {Promise<boolean>} `true` if the request was successful, false otherwise.
   */
  const removeProjectParticipant = async (projectId: number, projectParticipationId: number): Promise<boolean> => {
    const { status } = await axios.delete(`/api/project/${projectId}/participants/${projectParticipationId}/delete`);

    return status === 200;
  };

  /**
   * Update project participant role.
   *
   * @param {number} projectId
   * @param {number} projectParticipationId
   * @param {string} role
   * @return {*}  {Promise<boolean>}
   */
  const updateProjectParticipantRole = async (
    projectId: number,
    projectParticipationId: number,
    roleId: number
  ): Promise<boolean> => {
    const { status } = await axios.put(`/api/project/${projectId}/participants/${projectParticipationId}/update`, {
      roleId
    });

    return status === 200;
  };

  return {
    getAllUserProjectsForView,
    getProjectsList,
    createProject,
    getProjectForView,
    uploadProjectAttachments,
    uploadProjectReports,
    updateProjectReportMetadata,
    getProjectForUpdate,
    updateProject,
    getProjectAttachments,
    getAttachmentSignedURL,
    deleteProjectAttachment,
    deleteFundingSource,
    addFundingSource,
    deleteProject,
    publishProject,
    makeAttachmentSecure,
    makeAttachmentUnsecure,
    getProjectReportMetadata,
    getProjectParticipants,
    addProjectParticipants,
    removeProjectParticipant,
    updateProjectParticipantRole
  };
};

export default useProjectApi;

/**
 * Returns a set of supported api methods for working with public (published) project records.
 *
 * @param {AxiosInstance} axios
 * @return {*} object whose properties are supported api methods.
 */
export const usePublicProjectApi = (axios: AxiosInstance) => {
  /**
   * Get public facing (published) projects list.
   *
   * @return {*}  {Promise<IGetProjectsListResponse[]>}
   */
  const getProjectsList = async (): Promise<IGetProjectsListResponse[]> => {
    const { data } = await axios.get(`/api/public/projects`);

    return data;
  };

  /**
   * Get public (published) project details based on its ID for viewing purposes.
   *
   * @param {number} projectId
   * @return {*} {Promise<IGetProjectForViewResponse>}
   */
  const getProjectForView = async (projectId: number): Promise<IGetProjectForViewResponse> => {
    const { data } = await axios.get(`/api/public/project/${projectId}/view`);

    return data;
  };

  /**
   * Get public (published) project attachments based on project ID
   *
   * @param {number} projectId
   * @returns {*} {Promise<IGetProjectAttachmentsResponse>}
   */
  const getProjectAttachments = async (projectId: number): Promise<IGetProjectAttachmentsResponse> => {
    const { data } = await axios.get(`/api/public/project/${projectId}/attachments/list`);

    return data;
  };

  /**
   * Get public (published) project attachment S3 url based on project and attachment ID
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {string} attachmentType
   * @returns {*} {Promise<string>}
   */
  const getAttachmentSignedURL = async (
    projectId: number,
    attachmentId: number,
    attachmentType: string
  ): Promise<string> => {
    const { data } = await axios.get(`/api/public/project/${projectId}/attachments/${attachmentId}/getSignedUrl`, {
      params: { attachmentType: attachmentType },
      paramsSerializer: (params) => {
        return qs.stringify(params);
      }
    });

    return data;
  };

  /**
   * Get project report metadata based on project ID, attachment ID, and attachmentType
   *
   * @param {number} projectId
   * @param {number} attachmentId
   * @param {string} attachmentType
   * @returns {*} {Promise<string>}
   */
  const getPublicProjectReportMetadata = async (
    projectId: number,
    attachmentId: number
  ): Promise<IGetReportMetaData> => {
    const { data } = await axios.get(`/api/public/project/${projectId}/attachments/${attachmentId}/metadata/get`, {
      params: {},
      paramsSerializer: (params) => {
        return qs.stringify(params);
      }
    });

    return data;
  };

  return {
    getProjectsList,
    getProjectForView,
    getProjectAttachments,
    getAttachmentSignedURL,
    getPublicProjectReportMetadata
  };
};
