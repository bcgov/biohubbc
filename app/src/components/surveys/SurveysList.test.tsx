import { render } from '@testing-library/react';
import { IGetProjectSurvey } from 'interfaces/useProjectApi.interface';
import React from 'react';
import SurveysList from './SurveysList';

describe('SurveysList', () => {
  it('renders correctly with surveys', () => {
    const surveysList: IGetProjectSurvey[] = [
      {
        id: 1,
        name: 'Moose Survey 1',
        species: 'Moose',
        start_date: '2021-04-09 11:53:53',
        end_date: '2021-05-09 11:53:53',
        status_name: 'Unpublished'
      },
      {
        id: 2,
        name: 'Moose Survey 2',
        species: 'Moose',
        start_date: '2021-04-09 11:53:53',
        end_date: '2021-06-10 11:53:53',
        status_name: 'Published'
      }
    ];

    const { asFragment } = render(<SurveysList surveysList={surveysList} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no surveys', () => {
    const { asFragment } = render(<SurveysList surveysList={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
