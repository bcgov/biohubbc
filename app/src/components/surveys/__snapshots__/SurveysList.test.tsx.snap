// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveysList renders correctly with no surveys 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiTableContainer-root"
    >
      <table
        aria-label="surveys-list-table"
        class="MuiTable-root makeStyles-table-7"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-8"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-8"
              scope="col"
            >
              Species
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-8"
              scope="col"
            >
              Timeline
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-8"
              scope="col"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
              colspan="4"
            >
              No Surveys
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SurveysList renders correctly with surveys 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiTableContainer-root"
    >
      <table
        aria-label="surveys-list-table"
        class="MuiTable-root makeStyles-table-1"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-2"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-2"
              scope="col"
            >
              Species
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-2"
              scope="col"
            >
              Timeline
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head makeStyles-heading-2"
              scope="col"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <button
                class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiLink-button MuiTypography-body2 MuiTypography-colorPrimary"
              >
                Moose Survey 1
              </button>
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Moose
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              April 09, 2021 - May 09, 2021
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <div
                class="MuiChip-root makeStyles-chip-4 makeStyles-chipUnpublished-5"
              >
                <span
                  class="MuiChip-label"
                >
                  UNPUBLISHED
                </span>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              <button
                class="MuiTypography-root MuiLink-root MuiLink-underlineAlways MuiLink-button MuiTypography-body2 MuiTypography-colorPrimary"
              >
                Moose Survey 2
              </button>
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Moose
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              April 09, 2021 - June 10, 2021
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <div
                class="MuiChip-root makeStyles-chip-4 makeStyles-chipPublished-6"
              >
                <span
                  class="MuiChip-label"
                >
                  PUBLISHED
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          id="mui-10526"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-10526 mui-6689"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            id="mui-6689"
            role="button"
            tabindex="0"
          >
            5
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="5"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-2 of 2
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
